contracts/extensions/TokenSupplier.sol

```diff
@@ -0,0 +1,193 @@
+import "./ColonyExtension.sol";
+
+
+contract TokenSupplier is ColonyExtension {
+
+  uint256 constant ISSUANCE_PERIOD = 1 days;
+  bytes32 constant VOTING_REPUTATION = keccak256("VotingReputation");
+  bytes32 constant VOTING_HYBRID = keccak256("VotingHybrid");
+
+  // Events
+
+  event TokenSupplyCeilingSet(uint256 tokenSupplyCeiling);
+  event TokenIssuanceRateSet(uint256 tokenIssuanceRate);
+  event TokensIssued(uint256 numTokens);
+
+  // Storage
+
+  address token;
+  uint256 tokenSupplyCeiling;
+  uint256 tokenIssuanceRate;
+  uint256 lastIssue;
+  uint256 lastRateUpdate;
+
+  // Modifiers
+
+  modifier initialised() {
+    require(lastIssue > 0, "token-supplier-not-initialised");
+    _;
+  }
+
+  modifier onlyRoot() {
+    require(colony.hasUserRole(msg.sender, 1, ColonyDataTypes.ColonyRole.Root), "token-supplier-caller-not-root");
+    _;
+  }
+
+  modifier onlyContract() {
+    uint256 size;
+    address msgSender = msg.sender;
+    assembly { size := extcodesize(msgSender) }
+    require(size > 0, "token-supplier-caller-must-be-contract");
+    _;
+  }
+
+  // Public
+
+  /// @notice Returns the identifier of the extension
+  function identifier() public override pure returns (bytes32) {
+    return keccak256("TokenSupplier");
+  }
+
+  /// @notice Returns the version of the extension
+  function version() public override pure returns (uint256) {
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    require(address(colony) == address(0x0), "extension-already-installed");
+
+    colony = IColony(_colony);
+    token = colony.getToken();
+  }
+
+  /// @notice Called when upgrading the extension (currently a no-op)
+  function finishUpgrade() public override auth {}
+
+  /// @notice Called when deprecating (or undeprecating) the extension (currently a no-op)
+  function deprecate(bool _deprecated) public override auth {}
+
+  /// @notice Called when uninstalling the extension
+  function uninstall() public override auth {
+    selfdestruct(address(uint160(address(colony))));
+  }
+
+  /// @notice Initialise the extension, must be called before any tokens can be issued
+  /// @param _tokenSupplyCeiling Total amount of tokens to issue
+  /// @param _tokenIssuanceRate Number of tokens to issue per day
+  function initialise(uint256 _tokenSupplyCeiling, uint256 _tokenIssuanceRate) public onlyRoot {
+    require(lastIssue == 0, "token-supplier-already-initialised");
+
+    tokenSupplyCeiling = _tokenSupplyCeiling;
+    tokenIssuanceRate = _tokenIssuanceRate;
+    lastIssue = block.timestamp;
+    lastRateUpdate = block.timestamp;
+  }
+
+  /// @notice Update the tokenSupplyCeiling, cannot set below current tokenSupply
+  /// @param _tokenSupplyCeiling Total amount of tokens to issue
+  function setTokenSupplyCeiling(uint256 _tokenSupplyCeiling) public initialised onlyRoot onlyContract {
```

Comment:
If this is set below the current `totalSupply`, it prevents calling both `issueTokens` and `setTokenIssuanceRate`. It means that the issuance that can be claimed keeps accumulating until the ceiling is set above `totalSupply`, at which point a load of tokens can be claimed. Is this something we should give people the ability to work around?